ddtrace/tracer/tracer.go

```diff
@@ -95,6 +96,15 @@ type tracer struct {
 
 	// statsd is used for tracking metrics associated with the runtime and the tracer.
 	statsd statsdClient
+
+	// abandonedSpans holds a linked list of potentially abandoned spans for all traces.
+	abandonedSpans *list.List
+
+	// cIn takes newly created spans and adds them to abandonedSpans
+	cIn chan *span
+
+	// cOut signals for a finished span to be removed from abandonedSpans
+	cOut chan *span
```

Comment:
@hannahkm Looking around in ways to encapsulate these variables and their associated logic I found the `concentrator` type in stats.go. No need to replicate it, but I thought it was a good piece of code to read.